s/core.go

```diff
@@ -186,8 +187,9 @@ func (spec SupervisorSpec) start(parentCtx context.Context, parentName string) (
 			select {
 			case err := <-errCh:
 				return err
-			case <-terminateCh:
-				return nil
+			case err := <-terminateCh:
```

Comment:
Something bothers me about both of these just returning an error. Here, I do not see a difference between the two channels. Why both? I see that one if for the supervisor path and the other for the child path, but they are equivalent, essentially. Why both?

Reply:
Great catch, I think they are doing the same thing, I'm going to remove the `errCh` now